### DISCUSSION TITLE
kustomize-controller: ability to automatically health check all resources

### DISCUSSION THREAD

**Discussion by seaneagan** [fluxcd/flux2#324] [Proposals]
Currently one has to explicitly list all resources that they want to do a health check on. If one wants to health check all the resources and there are many of them, this is cumbersome and error prone. Thus it would be useful if there were a way to automatically do this. Also since health checks is optional, if someone forgets to set this field, they will not get any health checks and so may accidentally introduce race conditions. Thus I think having the default be to health check all resources would be ideal. If someone wanted no health checks they could easily set the field to `[]`.

*Implementation:* The labels that the kustomize-controller adds to each resource it applies *could* be used to lookup which resources to health check, but this would require API calls for each kubernetes API type (in addition to the ones kstatus will do), and I'm not sure if kubernetes supports only returning the names of resources in list calls, so this would be pretty inefficient. However, #309 would make this really easy since the inventory object already contains the full list of resources.

**Comment by stefanprodan** [comment]
I find the inventory object to be a waste of etcd space. The GC [snapshot](https://github.com/fluxcd/kustomize-controller/blob/main/api/v1beta1/snapshot_types.go) has enough info to select all objects originated from a particular reconciliation.

**Reply by seaneagan** [reply]
Thanks for the pointer, looks like the snapshot (along with applied labels) does indeed already account for the data in the inventory object and more.
